Pet-Clinic-Frontend/src/main.tsx

```diff
@@ -3,29 +3,58 @@ import ReactDOM from 'react-dom/client'
 import {
   createBrowserRouter,
   RouterProvider,
+  Outlet
 } from "react-router-dom"
 import './index.css'
 import Registration from './pages/Registration';
 import Login from './pages/Login';
 import Landing from './pages/Landing';
+import { AuthProvider } from './pages/AuthContext';
+import Navbar from "./components/Navbar"
+import { routes } from "./routes"
+import PrivateRoute from './pages/PrivateRoute';
+import ProfileEdit from './pages/ProfileEdit';
+
+const Root = () => (
+  <>
+    <Navbar />
+    <Outlet />
+  </>
+);
+
 
 const router = createBrowserRouter([
   {
-    path: "/",
-    element: <Landing />,
-  },
-  {
-    path: "/registration",
-    element: <Registration />,
-  },
-  {
-    path: "/login",
-    element: <Login />,
+    element: <Root />,
+    children: [
+      {
+        path: routes[0].path,
+        element: <Landing />,
+      },
+      {
+        path: routes[1].path,
+        element: <Registration />,
+      },
+      {
+        path: routes[2].path,
+        element: <Login />,
+      },
+      {
+        path: routes[3].path,
+        element: (
+         <PrivateRoute>
+            <ProfileEdit />
+        </PrivateRoute>
+        ),
+      },
```

Comment:
```suggestion
      ...routes.map(({ path, requiresAuth }) => ({
        path,
        element: requiresAuth ? (
          <PrivateRoute>
            <ProfileEdit />
          </PrivateRoute>
        ) : <ProfileEdit />,
      })),
```

Reply:
You can make create the router dynamically too.

Reply:
New code always ends in <ProfileEdit>.  It does not seem to me beneficial :-).  OK I understand  I will find out how to provide different component/pages.

Reply:
always ends in ProfileEdit

Reply:
Oh, yes you are correct, you should have a reference to the components too, which one should be rendered. This was just an example. THe type that you wanted to recreate is the https://reactrouter.com/en/main/routers/create-browser-router#type-declaration
type declaration from here.